test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -299,6 +309,105 @@ def test_full_with_dtype_overridden(self):
         expected_tensor = torch.full((h, w), fill_value=fill_value, device=local_device, dtype=torch.double)
         self.assertEqual(expected_tensor, local_tensor)
 
+
+class TestShardedTensorOps(ShardedTensorTestBase, MultiProcessTestCase):
+    def _run_sharded_linear(self, spec, input_size, linear_size, sharded_dim):
+        # Use same seed.
+        torch.manual_seed(0)
+        local_linear = torch.nn.Linear(*linear_size).cuda(self.rank)
+
+        sharded_linear = torch.nn.Linear(*linear_size)
+
+        # Copy the weights and bias from local linear
+        sharded_linear.weight = local_linear.weight
+        sharded_linear.bias = local_linear.bias
+
+        # Shard the parameter.
+        shard_parameter(sharded_linear, 'weight', spec)
+
+        # Run sharded computation
+        torch.manual_seed(self.rank)  # inputs different on each rank
+        inp = torch.rand(*input_size).cuda(self.rank)
+        sharded_output = sharded_linear(inp)
+
+        # Run local computation
+        local_output = local_linear(inp)
+
+        # Verify
+        self.assertEqual(local_output, sharded_output)
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_sharded_linear_colwise(self):
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:0/cuda:0",
+                "rank:1/cuda:1",
+                "rank:2/cuda:2",
+                "rank:3/cuda:3",
+            ],
+        )
+
+        self._run_sharded_linear(spec, [5, 17], [17, 12], 0)
+        self._run_sharded_linear(spec, [5, 21], [21, 11], 0)
+        self._run_sharded_linear(spec, [5, 23], [23, 13], 0)
+        self._run_sharded_linear(spec, [5, 15], [15, 14], 0)
+
+        # Test different ordering.
+        spec = ChunkShardingSpec(
+            dim=0,
+            placements=[
+                "rank:1/cuda:1",
+                "rank:0/cuda:0",
+                "rank:3/cuda:3",
+                "rank:2/cuda:2",
+            ],
+        )
+
+        self._run_sharded_linear(spec, [5, 17], [17, 12], 0)
+        self._run_sharded_linear(spec, [5, 21], [21, 11], 0)
+        self._run_sharded_linear(spec, [5, 23], [23, 13], 0)
+        self._run_sharded_linear(spec, [5, 15], [15, 14], 0)
+
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
```

Comment:
nit

probably add another two test cases:

1. test the torch_function dispatch for  torch.nn.linear(input, shared_linear) ?

2. test shard_parameter func explicitly